From 965c0c46fcec3580829a9d28c6e1af7fea5def87 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 19 May 2016 21:11:06 -0700
Subject: [PATCH 3/4] clang: Driver/tools.cpp: Add -lssp and -lssp_nonshared on
 musl

musl driver will need to add ssp libraries to linker cmdline
when using stack protector commands on compiler cmdline

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 lib/Driver/Tools.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index 1b4c469..b850b0f 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -9463,6 +9463,13 @@ void gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
 
       CmdArgs.push_back("-lc");
 
+      if (ToolChain.getTriple().isLinuxMuslEnvironment() &&
+          (Args.hasArg(options::OPT_fstack_protector) ||
+          Args.hasArg(options::OPT_fstack_protector_strong) ||
+          Args.hasArg(options::OPT_fstack_protector_all))) {
+        CmdArgs.push_back("-lssp_nonshared");
+        CmdArgs.push_back("-lssp");
+      }
       if (Args.hasArg(options::OPT_static))
         CmdArgs.push_back("--end-group");
       else
-- 
2.8.3

